string pb_type
byte[] encoded
